### DISCUSSION TITLE
Disable the `unicorn/prefer-string-replace-all` rule

### DISCUSSION THREAD

**Discussion by leroykorterink** [mediamonks/eslint-config#233] [General]
There is a proposal to disable the `unicorn/prefer-string-replace-all` rule.

> `unicorn/prefer-string-replace-all`: introduces unnecessary verbosity
> \- @edgardz

**Comment by leroykorterink** [comment]
We should not disable this rule because `String#replaceAll` is significantly faster than an implementation with `String#replace` where a regular expression with the global flag is used. A regular expression may also introduce security issues in some scenarios.

Please see [the documentation](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-string-replace-all.md) of the rule.

**Comment by ThaNarie** [comment]
> introduces unnecessary verbosity

How does it do that according to you?

```ts
// you prefer this?
string.replaceAll(/foo/g, 'bar');

// this is more verbose?
string.replaceAll('foo', 'bar');
```

To me it feels like the regular expression variants are more verbose than using `replaceAll`?

But maybe we're missing the point, so I would love some examples :)

**Reply by edgardz** [reply]
Minor issue. I was just trying not to enforce one more camelCased method floating in the code.

**Reply by ThaNarie** [reply]
I also think I made a mistake in my example.

**worst example**
typing 3 more chars
```ts
// you prefer
string.replace(/Works for u flag too/gu, '');

// rule requires replaceAll, to clearly indicate intent
string.replaceAll(/Works for u flag too/gu, '');
```

**useful example**
```ts
// you prefer
string.replace(/Works for u flag too/g, '');

// rule requires replaceAll, since it's way faster to execute
string.replaceAll('Works for u flag too', '');
```

Since this rule is auto-fixable, and reading those 3 characters makes the intent way more clearer, I would still be in favor of keeping this rule.
